### DISCUSSION TITLE
Waveshares - no link on Server?

### DISCUSSION THREAD

**Discussion by solarjones** [alienatedsec/solis-ha-modbus-cloud#6] [Q&A] [answered]
Hi,
Today I wired 2x Waveshare RS485 to POEs with some success, but feel I may have missed something in the config as the 'server' does not show a solid blue Link light. But everything appears to be working :) 

- Solis Cloud via the Solis App is working (needed a datalogger reboot (unplug/plug) as it initially hung after ~20mins)
- HA using SolaX Inverter Modbus integration appears to be up and running to (early days - just spotted it had set my inverter to Backup mode without my say so. Had to set it to Timed Charge/Discharge in HA).

My setup:
Solis Hybrid inverter with an S2-WL using LAN datalogger.
Waveshare - Server IP .202  (labelled GW .202 as this is the gateway) 
Waveshare - Client IP .203  (labelled DLG .203 as this is the datalogger)
HA - IP .235
Vircom - IP .120 

![image](https://user-images.githubusercontent.com/100320468/226736272-b931e685-7117-4426-b5b0-5fdd2106e1b2.png)

Vircom configs:
![image](https://user-images.githubusercontent.com/100320468/226734798-d9d49ffc-da59-4481-ba73-fb703cfda228.png)

Server: 
![image](https://user-images.githubusercontent.com/100320468/226734924-c88305cc-d781-47f5-9e51-bb274b423423.png)

Server advanced:
![image](https://user-images.githubusercontent.com/100320468/226735066-d632110c-6dbd-4a69-a07c-9a59b22c10fe.png)

Client:
![image](https://user-images.githubusercontent.com/100320468/226735186-93fe958c-9550-4bb7-b458-e9bfb56060c1.png)

Client advanced:
![image](https://user-images.githubusercontent.com/100320468/226735273-ee5b0cf0-f260-489b-a4bb-a4f199a74995.png)

In HA, I've successfully configured the integration and have telemetry coming through which is great. Pointed it at the Waveshare Server on .202. 
<img width="320" alt="Image 135" src="https://user-images.githubusercontent.com/100320468/226746020-14073169-3191-4816-bf76-033a4bbf7077.png">

<img width="240" alt="PastedGraphic-6 134" src="https://user-images.githubusercontent.com/100320468/226738613-09a38331-b3ae-4d0b-bee4-f1b1cbe97f96.png">

So, the question I've got is why don't both my Waveshares have the solid blue Link light on? 
Client: LINK = solid Blue
Client: ACT = flashes Blue with traffic
Server: LINK = solid Green
Server: ACT = occasionally flashes blue with traffic

Thanks for all your help, I would not have been brave or knowledgeable enough to try this otherwise!

**Comment by alienatedsec** [comment]
@solarjones Is your network liking Static IPs?

**Reply by solarjones** [reply]
Yes, playing nicely with other devices. My thinking was having DHCP allocate IPs could cause HA not to find the Waveshare Server. I assume that is the correct 'target' for the HA to use?

**Reply by alienatedsec** [reply]
@solarjones despite cabling, it should not be an issue unless you are using a mixed environment (DHCP Range +Static Range) If those are overlapping and you assigned the IP, which is in the DHCP range, it could be creating conflicts and disconnecting either device with the same IP.

**Comment by alienatedsec** [comment]
@solarjones is your data logger linked to the Server or Client?

**Reply by solarjones** [reply]
It's linked to the Client.  <img width="800" alt="image" src="https://user-images.githubusercontent.com/100320468/226754005-6824da0d-a76b-4aba-a646-c9b8b976714c.png">

**Reply by alienatedsec** [reply]
@solarjones potential problem

I noticed you are using two pairs of twisted cabling. For RS485 it's recommended to use a pair which is twisted but for both A and B on one twisted pair. So, your blue and white/blue twisted cables should be used for A(blue) and B(white/blue)

**Reply by solarjones** [reply]
Ah, ok, thats a rewire then :)  In my excitement I didn't realise thats how to do it.
Would that make the difference on the LINK status? If the signals were 'bad', wouldn't I see very odd data on both Solis Cloud and HA?

**Reply by alienatedsec** [reply]
@solarjones it is likely to create interference in noisy (electrical) environments and at long distances, if not wired the proper way. You can find out more about it on the web [or some quickly found details are available here.](https://www.analog.com/en/technical-articles/rs485-cable-specification-guide--maxim-integrated.html)

I also used earthing, but it may not be necessary.

**Reply by alienatedsec** [reply]
@solarjones I just realised that at the very beginning, my setup acted similar (green instead of the blue LED). Maybe worth considering leaving it running (HA + Datalogger) as is overnight and check it again in the morning.

**Reply by solarjones** [reply]
Overnight running of both HA + data logger didn't change LED status.

**Comment by alienatedsec** [comment, marked as answer]
@solarjones Another suggestion is to increase some values and match the settings according to my screenshots on the main page. It's also worth comparing the firmware version `V1.452` is definitely on both devices - I can't see it from your screenshots.

You could also swap them around and see if the same green LED is going to be on the other device if you change their roles and connect differently. You can easily unplug those green connectors and plug one into another without unscrewing wires.

**Reply by solarjones** [reply]
I increased the values using the exact same numbers as your screenshots. No change in LED status. Firmware versions are identical.

**Reply by solarjones** [reply]
I've just swapped the client/server units and reconfigured Waveshare. The lack of blue LED followed the unit. I'll leave it running today and if no change this evening, I think it could be a faulty LED light in one of the Waveshares.

**Reply by alienatedsec** [reply]
@solarjones that is unlucky - however, if everything works as expected, then you should be fine. You could order another one and return this faulty unit after receiving the replacement.

**Reply by solarjones** [reply]
Yep, Amazon are great!
![image](https://user-images.githubusercontent.com/100320468/227219976-840479d1-2b77-414c-bfc9-b88d207be63b.png)
2x LINK lights. All is well. Thanks for your guidance and help!

**Reply by solarjones** [reply]
Next improvement will be to add an earth from each Waveshare and install the Emergency Power Supply (or whatever its called) plug/cable to a double socket and back to earth.  Its a shame there isn't a function like the Tesla Gateway to allow the inverter/batteries to power the house during an outage, but wouldn't want to pay Telsa prices for the solution!

**Comment by magichew** [comment]
Sorry to jump in here, I haven't purchased the Waveshare devices but I have a question about the above screenshots. Are those screenshots taken from the device's web server or is there an application to install to get to the config?

**Reply by solarjones** [reply]
Hi - they are from the Windows based app (Vircom) which is required for the initial setup.  More info here [https://www.waveshare.com/wiki/RS485_TO_ETH_(B)](https://www.waveshare.com/wiki/RS485_TO_ETH_(B)
)

**Reply by alienatedsec** [reply]
@magichew those screenshots are from the app VirCom. It allows you to configure Waveshare devices even if they don't have an IP address assigned. You can also change some advanced settings, which are not available via the web interface.

**Reply by magichew** [reply]
Thank you for confirming that. Is the software purely for configuration and not something that will be regularly needed to make them work?

@alienatedsec will I need to configure advanced settings if I am following your set up?

It's not a problem to run a Windows VM, I just want to know what I'm getting myself in to.

**Reply by alienatedsec** [reply]
@magichew you only do the configuration once, and it's not needed to run in the background or even on a separate VM.

**Reply by magichew** [reply]
@alienatedsec That's great. Thank you.
